lib/encoding/yaml/parser.toit

```diff
@@ -0,0 +1,1689 @@
+              start-mark = mark
+              repeat: s-white
+              rest = string-since start-mark
+            return flatten-list_ [ folded, string.from-rune first, line, rest ]
+        return [folded]
+    return null
+
+  nb-ns-double-in-line -> string:
+    runes/List := []
+    repeat: 
+      white := repeat: s-white
+      rune := ns-double-char
+      if rune:
+        runes.add-all white
+        runes.add rune
+      rune
+    return string.from-runes runes
+
+  s-double-next-line n/int -> List?:
+    try-parse:
+      if breaks := s-double-break n:
+        try-parse:
+          if first-rune := ns-double-char:
+            line := nb-ns-double-in-line
+            first := "$(string.from-runes (flatten-list_ [breaks, first-rune]))$line"
+            rest/any := s-double-next-line n
+            if not rest:
+              mark := mark
+              repeat: s-white
+              rest = string-since mark
+            return flatten-list_ [ first, rest ]
+        return [string.from-runes breaks]
+    return null
+
+  s-double-break n/int -> List?:
+    if rune := s-double-esscaped n: return rune
+    if folded := s-flow-folded n:
+      runes := List
+      folded.do --runes: runes.add it
+      return runes
+    return null
+
+  s-double-esscaped n/int -> List?:
+    try-parse:
+      white-spaces := repeat: s-white
+      if match-char C-ESCAPE_ and
+         b-non-content and
+         (repeat: l-empty n FLOW-IN_) and
+         s-flow-line-prefix n:
+        return white-spaces
+    return null
+
+  ns-single-char -> int?:
+    try-parse:
+      if not s-white:
+        if rune := nb-single-char:
+          return rune
+    return null
+
+  nb-single-char -> int?:
+    try-parse:
+      if rune := c-quoted-quote: return '\''
+      if rune := nb-json:
+        if rune != C-SINGLE-QUOTE_: return rune
+    return null
+
+  ns-double-char -> int?:
+    try-parse:
+      if not s-white:
+        if rune := nb-double-char:
+          return rune
+    return null
+
+  nb-double-char -> int?:
+    if rune := c-ns-esc-char: return rune
+    try-parse:
+      if rune := nb-json:
+        if rune != C-ESCAPE_ and rune != C-DOUBLE-QUOTE_: return rune
+    return null
+
+  static SIMPLE-ESCAPES ::= { '0': 0, 'a': '\a', 'b': '\b', 't': '\t', '\t': '\t', 'n': '\n',
+                              'v': '\v', 'f': '\f', 'r': '\r', 'e': 0x18, ' ': ' ', '"': '"',
+                              '/': '/', '\\': '\\', 'N': 0x85, '_': 0x0a, 'L': 0x2028,
+                              'P': 0x2029 }
+  static SIMPLE-ESCAPE-KEYS ::= keys-as-set_ SIMPLE-ESCAPES
+
+  c-ns-esc-char -> int?:
+    try-parse:
+      if match-char C-ESCAPE_:
+        if rune := match-chars SIMPLE-ESCAPE-KEYS:
+          return SIMPLE-ESCAPES[rune]
+        if match-char 'x': if res := match-hex 2: return res
+        if match-char 'u':
+          if res := match-hex 4:
+            if 0xd800 <= res <= 0xdbff:
+              if part-2 := c-ns-esc-char:
```

Comment:
Where does this come from?
So YAML allows to write 16-bit code-points in a string?
The spec doesn't seem to mention it.
Add a comment?

Reply:
YAML is an extension of JSON, so I assume that it should support 16-bit surrogates. The spec is very silent about this.